wallet/src/de/schildbach/wallet/adapter/BuyAndSellDashServicesAdapter.kt

```diff
@@ -40,16 +46,26 @@ class BuyAndSellDashServicesAdapter( val config: Configuration,val onClickListen
                 binding.serviceBalance.setFormat(config.format.noCode())
                 binding.serviceBalance.setApplyMarkup(false)
                 binding.serviceBalance.setAmount(Coin.ZERO)
-                binding.serviceFiatAmount.setFormat(
-                    MonetaryFormat().noCode().minDecimals(2).code(
-                        0, config.exchangeCurrencyCode+Constants.PREFIX_ALMOST_EQUAL_TO
-                    )
+
+                val currencySymbol = (NumberFormat.getCurrencyInstance() as DecimalFormat).apply {
+                    currency = Currency.getInstance(config.exchangeCurrencyCode)
+                }.decimalFormatSymbols.currencySymbol
+
+                val fiatFormat = MonetaryFormat().noCode().minDecimals(2).optionalDecimals()
+
+                binding.serviceFiatAmount.text = Constants.PREFIX_ALMOST_EQUAL_TO+ view.context.getString(
+                    R.string.fiat_balance_with_currency, fiatFormat.format(Coin.ZERO),
+                    currencySymbol
                 )
+
                 if(data.balance!=null){
                     binding.serviceBalance.setAmount(data.balance)
                 }
                 if(data.localBalance!=null){
-                    binding.serviceFiatAmount.setAmount(data.localBalance)
+                    binding.serviceFiatAmount.text = Constants.PREFIX_ALMOST_EQUAL_TO+  view.context.getString(
+                        R.string.fiat_balance_with_currency, fiatFormat.format(data.localBalance),
+                        currencySymbol
```

Comment:
Whether a currency symbol goes before or after an amount depends on locale of the device, but here it seems like it is always going to be after the amount.

Reply:
Actually, I'm not sure if this is required given that we ignore this everywhere in the app except for the Enter Amount screens. @HashEngineering

Reply:
In the figma design it's before the amount

Reply:
Figma designs don't represent all the possible states of UI. Sometimes it has currency before the amount, sometimes after. Generally, it should depend on the device locale. NumberFormat class handles this automatically, but we probably need to use MonetaryFormat which doesn't.